hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.hedera.node.app.spi.info;
+
+import com.swirlds.common.system.address.AddressBook;
+
+/**
+ * Summarizes useful information about the nodes in the {@link AddressBook} from the Platform. In
+ * the future, there may be events that require re-reading the book; but at present nodes may treat
+ * the initializing book as static.
+ */
+public interface NodeInfo {
+
+    /**
+     * Convenience method to check if this node is zero-stake.
+     *
+     * @return whether this node has zero stake.
+     */
+    boolean isSelfZeroStake();
```

Comment:
NotNull annotation

Reply:
I see a pattern. Is Chat GPT making your code reviews? 🙂